Mathlib/Algebra/Star/Order.lean

```diff
@@ -155,9 +157,14 @@ theorem conjugate_nonneg {a : R} (ha : 0 ≤ a) (c : R) : 0 ≤ star c * a * c :
       _ ≤ star c * x * c + star c * y * c := add_le_add_left hy _
       _ ≤ _ := by rw [mul_add, add_mul]
 
+@[aesop safe apply]
 theorem conjugate_nonneg' {a : R} (ha : 0 ≤ a) (c : R) : 0 ≤ c * a * star c := by
   simpa only [star_star] using conjugate_nonneg ha (star c)
 
+@[aesop 90% apply (rule_sets := [CFC])]
+theorem conjugate_nonneg_of_isSelfAdjoint {a : R} (ha : 0 ≤ a) {c : R} (hc : IsSelfAdjoint c) :
+    0 ≤ c * a * c := by nth_rewrite 2 [← hc]; exact conjugate_nonneg' ha c
+
```

Comment:
You'll need to move `IsSelfAdjoint.mono` and `IsSelfAdjoint.of_nonneg` to the beginning of this section, which I think makes sense.
```suggestion
theorem IsSelfAdjoint.conjugate_nonneg {a : R} (ha : 0 ≤ a) {c : R} (hc : IsSelfAdjoint c) :
    0 ≤ c * a * c := by
  nth_rewrite 2 [← hc]; exact conjugate_nonneg' ha c

theorem conjugate_nonneg_of_nonneg {a : R} (ha : 0 ≤ a) {c : R} (hc : 0 ≤ c) :
    0 ≤ c * a * c :=
  IsSelfAdjoint.of_nonneg hc |>.conjugate_nonneg ha

```